src/pages/RFOX/hooks/useStakingInfoQuery.ts

```diff
@@ -0,0 +1,62 @@
+import { skipToken } from '@tanstack/react-query'
+import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import type { Address, ReadContractReturnType } from 'viem'
+import { getAddress } from 'viem'
+import { readContract } from 'viem/actions'
+import { arbitrum } from 'viem/chains'
+import type { Config } from 'wagmi'
+import { type ReadContractQueryKey, useQuery } from 'wagmi/query'
+import { viemClientByNetworkId } from 'lib/viem-client'
+
+type StakingInfoQueryKey = ReadContractQueryKey<
+  typeof foxStakingV1Abi,
+  'stakingInfo',
+  readonly [Address],
+  Config
+>
+type StakingInfo = ReadContractReturnType<typeof foxStakingV1Abi, 'stakingInfo', readonly [Address]>
+type UseStakingInfoQueryProps<SelectData = StakingInfo> = {
+  stakingAssetAccountAddress: string | undefined
+  select?: (stakingInfo: StakingInfo) => SelectData
+}
+const client = viemClientByNetworkId[arbitrum.id]
+
+export const useStakingInfoQuery = <SelectData = StakingInfo>({
+  stakingAssetAccountAddress,
+  select,
+}: UseStakingInfoQueryProps<SelectData>) => {
+  // wagmi doesn't expose queryFn, so we reconstruct the queryKey and queryFn ourselves to leverage skipToken type safety
+  const queryKey: StakingInfoQueryKey = [
+    'readContract',
+    {
+      address: RFOX_PROXY_CONTRACT_ADDRESS,
+      functionName: 'stakingInfo',
+      args: [stakingAssetAccountAddress ? getAddress(stakingAssetAccountAddress) : ('' as Address)],
```

Comment:
is there any way we can avoid this coalescing by making `stakingAssetAccountAddress` non-optional?

Reply:
Not sure if we mean non-optional at props-level or at queryKey level?

- At hook level we definitely don't want to make the prop required, narrowing is the whole goal of this wrapper
- At types level, we're honouring `wagmi/query`'s types i.e this guy

https://github.com/shapeshift/web/blob/06e83d905b37d2ae3cd92b822ed881dab202b310/src/pages/RFOX/hooks/useStakingInfoQuery.ts#L13-L18

Not concerned at alll about the cast in the `queryKey` since all we're concerned here is `queryFn` type-safety i.e this line:

https://github.com/shapeshift/web/blob/06e83d905b37d2ae3cd92b822ed881dab202b310/src/pages/RFOX/hooks/useStakingInfoQuery.ts#L54

Reply:
FYI that's what you get if you remove the type-safety in the `queryFn` - `queryKey` is just there as a companion because react-query and is totally decoupled from it 
<img width="1094" alt="image" src="https://github.com/shapeshift/web/assets/17035424/e020d789-978e-453b-bffd-5cc7288f3086">